grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -399,4 +399,46 @@ class CycleCountService {
             transaction.addToTransactionEntries(transactionEntry)
         }
     }
+
+    CycleCountItemDto updateCycleCountItem(CycleCountUpdateItemCommand command) {
+        command.cycleCountItem.quantityCounted = command.quantityCounted
+        command.cycleCountItem.comment = command.comment
+        command.cycleCountItem.countIndex = command.recount ? 1 : 0
+        command.cycleCountItem.status = CycleCountItemStatus.COUNTING
+
+        return command.cycleCountItem.toDto()
+    }
+
+    CycleCountItemDto createCustomCycleCountItem(CycleCountCustomItemCommand command) {
+        CycleCountItem cycleCountItem = new CycleCountItem(
+                facility: command.facility,
+                status: CycleCountItemStatus.COUNTING,
+                countIndex: command.recount ? 1 : 0,
+                quantityOnHand: command.quantityCounted, // FIXME: should we assume that qoh = quantityCounted for a custom row?
```

Comment:
a good question. I'd assume that because the system doesn't have that bin + lot for the product, that QoH would be 0

Reply:
I think that's fair, but a validation would probably be a good idea if we were worried about a "race condition" i.e. someone added the new inventory item after we started the count.